ext_mongodb.php

```diff
@@ -718,17 +775,82 @@ function __debugInfo() : array;
 
 final class Javascript implements Type, \Serializable
 {
-	use DenySerialization;
-
 	private $code;
 
-	public function __construct(string $code, private ?mixed $scope = NULL)
+	static private function checkArray(array $state)
+	{
+		if (
+			!array_key_exists( 'code', $state ) || !is_string( $state['code'] )
+		) {
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "MongoDB\BSON\Javascript initialization requires \"code\" string field" );
+		}
+
+		if ( array_key_exists( 'scope', $state ) && ( $state['scope'] !== NULL ) )
+		{
+			if ( !is_array( $state['scope'] ) && !is_object( $state['scope'] ) )
+			{
+				$valueType = gettype( $state['scope'] );
+				throw new \MongoDB\Driver\Exception\InvalidArgumentException( "Expected scope to be an array or object, {$valueType} given" );
+			}
+		}
+	}
+
+	public function serialize() : string
+	{
+		$s = [];
+		$s['code'] = $this->code;
+		if ( isset( $this->scope ) )
+		{
+			$s['scope'] = $this->scope;
+		}
+		return serialize( $s );
+	}
+
+	public function unserialize(mixed $serialized) : void
+	{
+		$unserialized = unserialize( $serialized );
+		self::checkArray( $unserialized );
+		if ( array_key_exists( 'scope', $unserialized ) )
```

Comment:
IIRC, PHPC always included the "scope" field when serializing, so it could be `null`. `php_phongo_javascript_init()` only attempts to do something with it if it's an array of object. Perhaps this is a fault of PHPC and we should raise an error if "scope" is not null or an array or object (e.g. throw an exception if it's an integer).